hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ECUnderReplicationHandler.java

```diff
@@ -315,14 +315,8 @@ private int processMissingIndexes(
                 sourceDatanodesWithIndex, selectedDatanodes,
                 int2byte(missingIndexes),
                 repConfig);
-        replicationManager.sendDatanodeCommand(reconstructionCommand,
-            container, selectedDatanodes.get(0));
-        // For each index that is going to be reconstructed with this command,
-        // adjust the replica count to reflect the pending operation.
-        for (int i = 0; i < missingIndexes.size(); i++) {
```

Comment:
This, I believe, was an existing bug in the handler. The pendingOps are adjusted via the `replicationManager.sendDatanodeCommand()`, so they were being adjust here and then again in RM. I am not sure if that would have caused a problem or not.

Reply:
This code was not adjusting the global pending ops. It was only adjusting the local copy. I change I made only a week or two ago. I will restore it, as it has caused a failing test which luckily caught the fact it should not be removed!